src/python/grpcio/grpc/__init__.py

```diff
@@ -299,6 +299,23 @@ def add_callback(self, callback):
     raise NotImplementedError()
 
 
+############################  gRPC Metadata  ##################################
+
+
+class Metadatum(six.with_metaclass(abc.ABCMeta, tuple)):
+  """A single entry in a Metadata collection.
+  
+  A metadata entry is a 2-tuple where the first entry is the key and the
+  second is the value, i.e. (key, value).  The metadata key is an ASCII str,
+  and must be a valid HTTP header name.  The metadata value can be
+  either a valid HTTP ASCII str, or bytes.  If bytes are provided,
+  the key must end with '-bin', i.e. ('binary-metadata-bin', b'\x00\xFF')
+
+  Metadatum is an abstract type, all instances should be tuples as 
+  described above.
+  """
```

Comment:
See above.  Clients providing tuples as metadata would be invalid according to the type spec.